2_Services.Tests/Base/ServiceTest.cs

```diff
@@ -0,0 +1,189 @@
+using Moq;
+using NUnit.Framework;
+using Repositories.Base;
+using Services.Base;
+using System;
+using System.Collections.Generic;
+using System.Reflection;
+using System.Threading.Tasks;
+
+namespace Services.Tests.Base
+{
+    /// <summary>
+    /// Base test class which should be inherited from when creating unittests for the services.
+    /// By inheriting, all the default tests are included in the new test class.
+    /// YOU SHOULD OVERRIDE THE DEFAULT TEST TO ADD THE [Test] attribute.
+    /// If you do not override the tests and add the [Test] attribute, the default tests will not be triggered.
+    /// </summary>
+    /// <typeparam name="TDomain">Modelclass which is used to test</typeparam>
+    /// <typeparam name="TService">Service which should be tested</typeparam>
+    /// <typeparam name="TRepository">IRepository which is used to mock the used Repository</typeparam>
+    public abstract class ServiceTest<TDomain, TService, TRepository>
+        where TDomain : class 
+        where TService: class, IService<TDomain>
+        where TRepository: class, IRepository<TDomain>
+    {
+        protected TService Service;
+        protected Mock<TRepository> RepositoryMock;
+
+        /// <summary>
+        /// Initialize runs before every test
+        /// Mock the repository given as generic in TRepository
+        /// Initialize the service with reflection
+        /// </summary>
+        [SetUp]
+        public void Initialize() 
+        {
+            // Mock the repository
+            RepositoryMock = new Mock<TRepository>();
+
+            // Create the service with reflection
+            Type serviceType = typeof(TService);
+            ConstructorInfo serviceCtor = serviceType.GetConstructor(new[] { typeof(TRepository) });
+            Service = (TService)serviceCtor.Invoke(new object[] { RepositoryMock.Object });
+        }
+
+        /// <summary>
+        /// Test if the repository method is called and check if anything has changed to the entity
+        /// </summary>
+        /// <param name="entity">The entity which is used as data to test</param>
+        /// <returns></returns>
+        public virtual async Task FindAsyncTest_GoodFlow(TDomain entity)
+        {
+            RepositoryMock.Setup(
+                repository => repository.FindAsync(1))
+                .Returns(
+                    Task.FromResult(entity)
+                );
+
+            TDomain retrievedEntity = await Service.FindAsync(1);
+
+            Assert.DoesNotThrow(() => {
+                RepositoryMock.Verify(repository => repository.FindAsync(1), Times.Once);
+            });
+
+            Assert.AreEqual(entity, retrievedEntity);
+        }
+
+        /// <summary>
+        /// Test if the repository method is called
+        /// </summary>
+        /// <param name="entity">The entity which is used as data to test</param>
```

Comment:
Pretty obvious param comment. Is this necessary?

Reply:
The coding guidelines require a documentation comment for each method parameter.